### DISCUSSION TITLE
How to open a .app on macOs without errors

### DISCUSSION THREAD

**Discussion by Pal3004** [airsdk/Adobe-Runtime-Support#3299] [General]
Hello everyone, 

I have an .air application that I want to use on macOS Sonoma.
 I managed to bundle it into a .app format, but when I provide a DMG or PKG version of this app to someone, macOS displays the following errors:
![Capture d'écran 2024-06-12 110713](https://github.com/airsdk/Adobe-Runtime-Support/assets/98019813/2d3352c8-0691-422e-ae38-447b66f362cb)
![Capture d'écran 2024-06-12 110742](https://github.com/airsdk/Adobe-Runtime-Support/assets/98019813/4bf9168c-6c22-4308-8e30-11338e18cee3)
The .app itself also shows the same errors. 
I know how to ignore them, but do you know if there is a solution to open the app without using the terminal or complicated manipulations because the app will be distributed and I would like the easiest way to execute it and download it. 

Thanks in advance for your help!

**Comment by ajwfrost** [comment]
If you've got the .app then you would need to get that properly code-signed, using a Developer ID Application certificate.
If you want to distribute this within a .dmg or .pkg file, you will need to sign that package using a Developer ID Installer certificate, and then notarize it.

Assuming you've got those certificates set up properly within the Apple developer portal and imported into your keychain, then signing can be done via ADT:
```
adt -sign -storetype KeychainStore -alias "Developer ID Application ....." -target bundle unsigned.app signed.app
```

The rest of it should follow public/general guidelines for how to package/sign/notarize an app. But in case it helps, below is a link to an earlier thread with some manual steps to getting everything done. I think "altool" is no longer used, there's a new "notarytool" - per the comment after this one:
https://github.com/airsdk/Adobe-Runtime-Support/discussions/2088#discussioncomment-3622583

thanks

**Reply by Pal3004** [reply]
Thanks for your response, 
I have signed my app with the Developer ID application and my .pkg and .dmg file too with my Developer ID installer. I ask apple to notarise my .pkg with the xcrun command and notarizetool but after 5 days I have no responses and a status invalid.
there is no other way to make this app distributable ?

**Reply by ajwfrost** [reply]
> I ask apple to notarise my .pkg with the xcrun command and notarizetool but after 5 days I have no responses 

Not sure what you're doing here but it should be automatic and take a matter of minutes!

A basic command-line would be:
```
xcrun notarytool submit PKGFILE --team-id ABCDEF1234 --apple-id apple-email@address.com --password abcd-efgh-ijkl-mnop --wait
```

See https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow#3087734
